src/core/shaders/graphs/metallicroughness.frag.json

```diff
@@ -2117,15 +2117,15 @@
                         "major": 2,
                         "minor": 0
                     },
-                    "substitution": "highp $type $output; if ($condition) discard; $output = $value;"
+                    "substitution": "highp $type $output = $value; if ($condition) discard;"
```

Comment:
With your last changes on the shader generator, isnt it more efficient to leave the assignment (which is going to perform the computation) after checking if we need to discard?

Reply:
afaik I had to change as the shader generator expect something like * var = assignment; *
So had to change the expression to make it work though I suspect this might result in the discard part completely being removed. Which would explain why the opacity is not working properly recently

Reply:
Fixed the discard if. Any substituion that requires more than one line or statement (more than one ;) should call a function. The qshadergenerator might not read it correctly or optimize it away when it shouldn't.

Reply:
nice. probably should be documented on the node editor and shader builder in qtbase.